contracts/ethregistrar/BulkRenewal.sol

```diff
@@ -3,46 +3,67 @@ pragma experimental ABIEncoderV2;
 
 import "../registry/ENS.sol";
 import "./ETHRegistrarController.sol";
+import "./IETHRegistrarController.sol";
 import "../resolvers/Resolver.sol";
+import "./IBulkRenewal.sol";
 
-contract BulkRenewal {
-    bytes32 constant private ETH_NAMEHASH = 0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae;
-    bytes4 constant private REGISTRAR_CONTROLLER_ID = 0x018fac06;
-    bytes4 constant private INTERFACE_META_ID = bytes4(keccak256("supportsInterface(bytes4)"));
-    bytes4 constant public BULK_RENEWAL_ID = bytes4(
-        keccak256("rentPrice(string[],uint)") ^
-        keccak256("renewAll(string[],uint")
-    );
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
 
-    ENS public ens;
+contract BulkRenewal is IBulkRenewal {
+    bytes32 private constant ETH_NAMEHASH =
+        0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae;
 
-    constructor(ENS _ens) public {
+    ENS public immutable ens;
+
+    constructor(ENS _ens) {
         ens = _ens;
     }
 
-    function getController() internal view returns(ETHRegistrarController) {
+    function getController() internal view returns (ETHRegistrarController) {
         Resolver r = Resolver(ens.resolver(ETH_NAMEHASH));
-        return ETHRegistrarController(r.interfaceImplementer(ETH_NAMEHASH, REGISTRAR_CONTROLLER_ID));
+        return
+            ETHRegistrarController(
+                r.interfaceImplementer(
+                    ETH_NAMEHASH,
+                    type(IETHRegistrarController).interfaceId
+                )
+            );
     }
 
-    function rentPrice(string[] calldata names, uint duration) external view returns(uint total) {
+    function rentPrice(string[] calldata names, uint256 duration)
+        external
+        view
+        override
+        returns (uint256 total)
+    {
         ETHRegistrarController controller = getController();
-        for(uint i = 0; i < names.length; i++) {
-            total += controller.rentPrice(names[i], duration);
+        for (uint256 i = 0; i < names.length; i++) {
+            (uint256 basePrice, ) = controller.rentPrice(names[i], duration);
```

Comment:
Won't this return the wrong value? It seems like it should sum the premiums as well, either as part of the total or as a separate field.

Reply:
Names that aren't expired don't have premiums?

Reply:
Excellent point. Perhaps we could sum it, or require() that it's zero, though? We don't want a situation where we use the premium for something and accidentally forget to charge it.

Reply:
Happy to sum it since it will just be 0. Leaning towards summing than require it is 0, since our tests should show a premium being incorrectly calculated if we change the premium oracle and accidentally let a premium slip into renews